sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -192,6 +192,17 @@ case class InsertIntoHiveTable(
     if (partition.nonEmpty) {
       if (numDynamicPartitions > 0) {
         if (overwrite && table.tableType == CatalogTableType.EXTERNAL) {
+          val numWrittenParts = writtenParts.size
+          val maxDynamicPartitionsKey = "hive.exec.max.dynamic.partitions"
+          val maxDynamicPartitions = hadoopConf.getInt(maxDynamicPartitionsKey, 1000)
+          if (numWrittenParts > maxDynamicPartitions) {
+            val maxDynamicPartitionsErrMsg =
+              s"Number of dynamic partitions created is $numWrittenParts" +
+                s", which is more than $maxDynamicPartitions" +
+                s". To solve this try to set $maxDynamicPartitionsKey" +
+                s" to at least $numWrittenParts."
+            throw new SparkException(maxDynamicPartitionsErrMsg)
```

Comment:
nit: we may want to group this error message and define it in `QueryExecutionErrors`.